### PR TITLE
[Experimental] Application Commands Module Fix

### PR DESCRIPTION
**Pull request by Lulalaby**

### CODE REVIEW

**Comment by JBraunsmaJr** [comment]
What exactly was this fixing?

To be honest I've been contemplating how we could improve the whole command system. 

For instance, the attribute-soup that happens when defining parameters for a command. It's not pretty to look at and you kinda.... lose where the actual parameter is lol.

**Comment by Lulalaby** [comment]
I've removed the part where it runs the registration in `_ = Task.Run(async () =>`. 
It somehow often caused problems.
`UpdateAsync` now awaits every registration as to see here: https://github.com/Aiko-IT-Systems/DisCatSharp/pull/135/files#diff-5191f86bd52548d384bc391ff817756f72f8dd975fe7c0b9b954c07b6c5cceefR433

**Comment by JBraunsmaJr** [comment]
> I've removed the part where it runs the registration in `_ = Task.Run(async () =>`. It somehow often caused problems. `UpdateAsync` now awaits every registration as to see here: https://github.com/Aiko-IT-Systems/DisCatSharp/pull/135/files#diff-5191f86bd52548d384bc391ff817756f72f8dd975fe7c0b9b954c07b6c5cceefR433

Should have put that in the description part of the PR lol.

**Comment by Lulalaby** [comment]
true, sorry, I'm extremely exhausted and tired..

File: DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs [hunk 1]
```diff
@@ -1713,7 +1716,7 @@ public async Task<IEnumerable<DiscordApplicationCommandAutocompleteChoice>> Prov
 		}
 
 		[SlashCommand("help", "Displays command help")]
-		internal async Task DefaultHelpAsync(InteractionContext ctx,
+		internal static async Task DefaultHelpAsync(InteractionContext ctx,
```
Comment (JBraunsmaJr) [inline DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs:1719]: why are we making this static?

Comment (Lulalaby) [reply DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs:1719]: VS suggested it

File: DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs [hunk 2]
```diff
@@ -1176,7 +1179,7 @@ private async Task<List<object>> ResolveInteractionCommandParameters(Interaction
 		/// </summary>
 		/// <param name="method">The method info.</param>
 		/// <param name="context">The base context.</param>
-		private async Task RunPreexecutionChecksAsync(MethodInfo method, BaseContext context)
+		private static async Task RunPreexecutionChecksAsync(MethodInfo method, BaseContext context)
```
Comment (JBraunsmaJr) [inline DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs:1182]: Why static

File: DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs [hunk 3]
```diff
@@ -152,8 +152,14 @@ internal static LogLevel ApplicationCommandsLogLevel
 		/// </summary>
 		internal static bool CheckAllGuilds { get; set; }
 
+		/// <summary>
+		/// Gets whether the registration check should be manually overriden.
+		/// </summary>
 		internal static bool ManOr { get; set; }
```
Comment (JBraunsmaJr) [inline DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs:158]: I never noticed this name before. Should we uh... call it ManualOverride or something. This member name is not super helpful. Granted the summary does help.

Comment (Lulalaby) [reply DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs:158]: It is really just internal. I don't want to expose it that much.

File: DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs (hunk 1 above)
Comment (JBraunsmaJr) [reply DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs:1719]: I would like to point out that not everything VS Suggests is gold. lol

Comment (Lulalaby) [reply DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs:1719]: fair

Comment (Lulalaby) [inline DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs:1719]:
```suggestion
		internal async Task DefaultHelpAsync(InteractionContext ctx,
```

File: DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs (hunk 2 above)
Comment (Lulalaby) [inline DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs:1182]:
```suggestion
		private async Task RunPreexecutionChecksAsync(MethodInfo method, BaseContext context)
```

File: DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs [hunk 4]
```diff
@@ -1002,7 +1005,7 @@ internal async Task RunCommandAsync(BaseContext context, MethodInfo method, IEnu
 			// Context menus
 			if (context is ContextMenuContext contextMenuContext)
 			{
-				await this.RunPreexecutionChecksAsync(method, contextMenuContext);
+				await RunPreexecutionChecksAsync(method, contextMenuContext);
```
Comment (Lulalaby) [inline DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs:1008]:
```suggestion
				await this.RunPreexecutionChecksAsync(method, contextMenuContext);
```

File: DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs [hunk 5]
```diff
@@ -986,7 +989,7 @@ internal async Task RunCommandAsync(BaseContext context, MethodInfo method, IEnu
 			// Slash commands
 			if (context is InteractionContext slashContext)
 			{
-				await this.RunPreexecutionChecksAsync(method, slashContext);
+				await RunPreexecutionChecksAsync(method, slashContext);
```
Comment (Lulalaby) [inline DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs:992]:
```suggestion
				await this.RunPreexecutionChecksAsync(method, slashContext);
```

**Pull request merged**